### ISSUE TITLE
Missing cost for ECS Service when task definition is used from a module and family is used instead of ARN

### ISSUE THREAD

**Issue by anitakrueger** [infracost/infracost#1915]
I was able to reproduce this and made a sample in https://github.com/anitakrueger/infracost-test

It comes down to infracost not being able to make the link between the ecs service and the task definition when the task definition is in a module and the task_definition parameter is set to the family instead of the arn.
This does seem to work when the task definition and ecs service are in the same terraform. Then switching to the family, infracost can still make the link.
In our case we set the task definition as family because we want the latest revision in terraform

```
resource "aws_ecs_service" "service" {
  name                               = "infracost-test"
==> WORKS  task_definition                    = module.task.arn
  cluster                            = aws_ecs_cluster.ecs_cluster.arn
  platform_version                   = "LATEST"
  launch_type                        = "FARGATE"
  propagate_tags                     = "TASK_DEFINITION"
  desired_count                      = 1
  deployment_maximum_percent         = 100
  deployment_minimum_healthy_percent = 0
  network_configuration {
    subnets          = ["subnet-0f6c2ff31a59db1d3"]
    assign_public_ip = false
  }
}

resource "aws_ecs_service" "service-nocost" {
  name                               = "infracost-test"
==> BREAKS  task_definition                    = module.task.family
  cluster                            = aws_ecs_cluster.ecs_cluster.arn
  platform_version                   = "LATEST"
  launch_type                        = "FARGATE"
  propagate_tags                     = "TASK_DEFINITION"
  desired_count                      = 1
  deployment_maximum_percent         = 100
  deployment_minimum_healthy_percent = 0
  network_configuration {
    subnets          = ["subnet-0f6c2ff31a59db1d3"]
    assign_public_ip = false
  }
}
```

The cost output shows 2 estimated resources, but there is only cost for one of them:
```
infracost-test % infracost breakdown --path . --show-skipped        
Evaluating Terraform directory at .
  ✔ Downloading Terraform modules 
  ✔ Evaluating Terraform directory 
  ✔ Retrieving cloud prices to calculate costs 

Project: github.com/anitakrueger/infracost-test.git

 Name                     Monthly Qty  Unit  Monthly Cost 
                                                          
 aws_ecs_service.service                                  
 ├─ Per GB per hour               0.5  GB           $1.62 
 └─ Per vCPU per hour            0.25  CPU          $7.39 
                                                          
 OVERALL TOTAL                                      $9.01 
──────────────────────────────────
4 cloud resources were detected:
∙ 2 were estimated
∙ 2 were free:
  ∙ 1 x aws_ecs_cluster
  ∙ 1 x aws_ecs_task_definition
```

The debug log from infracost shows this:
```
time="2022-07-27T00:54:03+01:00" level=debug routine=15 project_name= provider=terraform_dir block_name=module.task.aws_ecs_task_definition.task project_path=. attribute_name=memory msg="could not create references for expression type: &{{{{{} %!s(cty.primitiveTypeKind=78)}} %!s(*big.Float=&{512 0 0 1 false [9223372036854775808] 10})} modules/task-def/main.tf:6,30-33}"
time="2022-07-27T00:54:03+01:00" level=debug provider=terraform_dir block_name=module.task.aws_ecs_task_definition.task project_path=. attribute_name=id routine=15 project_name= msg="could not create references for expression type: &{{{{{} %!s(cty.primitiveTypeKind=83)}} 61f3e378-ae79-4cc0-b41c-d3430a90ad3e} :0,0-0}"
time="2022-07-27T00:54:03+01:00" level=debug block_name=module.task.aws_ecs_task_definition.task project_path=. routine=15 project_name= provider=terraform_dir attribute_name=arn msg="could not create references for expression type: &{{{{{} %!s(cty.primitiveTypeKind=83)}} arn:aws:hcl::4606465e-883c-4faf-987a-e6d71fb48652} :0,0-0}"
time="2022-07-27T00:54:03+01:00" level=debug provider=terraform_dir block_name=module.task.aws_ecs_task_definition.task attribute_name=cpu project_path=. routine=15 project_name= msg="could not create references for expression type: &{{{{{} %!s(cty.primitiveTypeKind=78)}} %!s(*big.Float=&{512 0 0 1 false [9223372036854775808] 9})} modules/task-def/main.tf:5,30-33}"
```

**Comment by aliscott** [comment]
I've added a fix for this specific case in #1917.

This fixes it for cases where the user specifies the family:
```hcl
resource "aws_ecs_service" "my_service" {
  ...
  task_definition = "<family>"
  ...
}
```

but it doesn't work if the user also specifies a revision, i.e:
```hcl
resource "aws_ecs_service" "my_service" {
  ...
  task_definition = "<family>:<revision>"
  ...
}
```

Fixing this final bit will probably require a bit of a refactor with how we handle references, so will be a more involved task.

**Comment by aliscott** [comment]
@anitakrueger this is now fixed in the [v0.10.9](https://github.com/infracost/infracost/releases/tag/v0.10.9) release, so you can update your action to point to the `infracost/actions/setup@v2` again now.

Thanks for your help debugging this issue 🙏. It was one of the best formed bug reports we have received since it had a fully reproducible example we could clone.
